writeengine/dictionary/we_dctnry.cpp

```diff
@@ -1111,14 +1111,14 @@ int Dctnry::insertDctnry(const int& sgnature_size,
     for (i = m_lastFbo; i < m_numBlocks; i++)
     {
         // @bug 3960: Add MAX_OP_COUNT check to handle case after bulk rollback
-        if ( ((m_freeSpace >= size) ||
-                ((size > 8176) && (m_freeSpace > m_totalHdrBytes))) &&
+        if ( ((m_freeSpace - HDR_UNIT_SIZE >= size) ||
+                ((size > 8176) && (m_freeSpace > HDR_UNIT_SIZE))) &&
                 (m_curOp    <  (MAX_OP_COUNT - 1)) )
         {
             // found the perfect block; signature size fit in this block
-            if (size > (m_freeSpace - m_totalHdrBytes))
+            if (size > (m_freeSpace - HDR_UNIT_SIZE))
```

Comment:
only need to make sure hdr fits since m_freeSpace has already taken the header into account